### DISCUSSION TITLE
How does nuxt realize the statistics of Unique Visitor and page view

### DISCUSSION THREAD

**Discussion by Abandoned9** [nuxt/nuxt#19543] [Questions] [answered]
As the title says, my server cannot access the external network, but only one web port is open. Therefore, third-party tools cannot be used. My API server is a SpringBoot, how can I achieve statistics on Unique Visitor and page view on Nuxt

**Comment by TheAlexLichter** [comment, marked as answer]
You could:

1. Use Client-Side tools like Plausible, Fathom, Google Analytics to track
2. Send a heartbeat call to your API server on init + route change and evaluate ip + page there to "build your own analytics".
